server/src/test/java/org/opensearch/common/settings/WriteableSettingTests.java

```diff
@@ -0,0 +1,487 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.common.settings;
+
+import org.junit.Before;
+import org.opensearch.Version;
+import org.opensearch.common.SuppressForbidden;
+import org.opensearch.common.bytes.BytesReference;
+import org.opensearch.common.io.stream.BytesStreamInput;
+import org.opensearch.common.io.stream.BytesStreamOutput;
+import org.opensearch.common.unit.ByteSizeUnit;
+import org.opensearch.common.unit.ByteSizeValue;
+import org.opensearch.common.unit.TimeValue;
+import org.opensearch.test.OpenSearchTestCase;
+
+import java.io.IOException;
+import java.lang.reflect.Field;
+import java.util.EnumMap;
+import java.util.EnumSet;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Function;
+
+import static org.opensearch.common.settings.Setting.Property;
+import static org.opensearch.common.settings.WriteableSetting.SettingType;
+
+public class WriteableSettingTests extends OpenSearchTestCase {
+
+    // These settings have a default value and null fallback
+    private final Map<SettingType, Setting<?>> settingMap = new EnumMap<>(SettingType.class);
+    // These settings have a fallback setting instead of a default
+    private final Map<SettingType, Setting<?>> settingWithFallbackMap = new EnumMap<>(SettingType.class);
```

Comment:
Nit: Do we need another `map` here? Can we add a differentiator in the key?

Reply:
> Can we add a differentiator in the key for fallback? Feel free to ignore

It's an EnumMap so no.... unless we switch to a String-based map or make they key a `Pair<SettingType, Boolean>` ... except for the `Version` type it wouldn't need the boolean ...

Also we test them separately so I think the existing setup works.

So I will "feel free to ignore" although it's a great question. :)